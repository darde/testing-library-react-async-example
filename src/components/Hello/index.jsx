import React, { useState, useEffect } from 'react';
import axios from 'axios';

const BASE_URL = 'https://jsonplaceholder.typicode.com/posts';

export default function Hello() {
  const [posts, setPosts] = useState([]);
  
  
  useEffect(() => {
    let mounted = true;

    const fetchData = async () => {
      const response = await axios.get(BASE_URL);
      if (mounted) {
        console.log(response.data);
        setPosts(response.data);
      }
    }
    fetchData();

    return () => {
      mounted = false;
    }
  }, []);
  console.log('length: ', posts.length);

  if(posts.length === 0) {
    console.log('returning... ');
    return <div>Loading...</div>
  }

  return (
    <div>
      <ul>
        {
          posts.map(
            post => <li key={post.id}>{post.title}</li>
          )
        }
      </ul>
    </div>
  )
}