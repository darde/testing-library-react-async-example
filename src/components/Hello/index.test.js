import React from 'react';
import {render, cleanup, waitForElement, act } from '@testing-library/react';
import mockAxios from 'axios';
import Hello from '.';

it('show loading message', () => {
  mockAxios.get.mockImplementationOnce(() => new Promise());
  const { getByText } = render(<Hello />);

  expect(getByText(/Loading/i)).toBeInTheDocument();
});

xit('renders hello correctly', async () => {
  mockAxios.get.mockResolvedValue({
    data: [
        { id: 1, title: 'image one' },
        { id: 2, title: 'image two' },
      ],
  });

  const { asFragment } = await waitForElement(() => render(<Hello />));
  
  expect(asFragment()).toMatchSnapshot();
});