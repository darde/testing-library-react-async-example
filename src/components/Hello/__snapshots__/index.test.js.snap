// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders hello correctly 1`] = `
<DocumentFragment>
  <h1>
    <ul>
      <li>
        image one
      </li>
      <li>
        image two
      </li>
    </ul>
  </h1>
</DocumentFragment>
`;
